src/x/ident/bytes_id.go

```diff
@@ -58,3 +58,55 @@ func (v BytesID) IsNoFinalize() bool {
 // Finalize is a no-op for a bytes ID as it has no associated pool.
 func (v BytesID) Finalize() {
 }
+
+var _ ID = &ReuseableBytesID{}
+
+// ReuseableBytesID is a reuseable bytes ID, use with extreme care in
+// places where the lifecycle is known (there is no checking with this
+// ID).
+type ReuseableBytesID struct {
+	bytes []byte
+}
+
+// NewReuseableBytesID returns a new reuseable bytes ID, use with extreme
+// care in places where the lifecycle is known (there is no checking with
+// this ID).
+func NewReuseableBytesID() *ReuseableBytesID {
+	return &ReuseableBytesID{}
+}
+
+// Reset resets the bytes ID for reuse, make sure there are zero references
+// to this ID from any other data structure at this point.
+func (i *ReuseableBytesID) Reset(bytes []byte) {
+	i.bytes = bytes
+}
+
+// Bytes implements ID.
+func (i *ReuseableBytesID) Bytes() []byte {
+	return i.bytes
+}
+
+// Equal implements ID.
+func (i *ReuseableBytesID) Equal(value ID) bool {
+	return bytes.Equal(i.bytes, value.Bytes())
+}
+
+// NoFinalize implements ID.
+func (i *ReuseableBytesID) NoFinalize() {
+}
+
+// IsNoFinalize implements ID.
+func (i *ReuseableBytesID) IsNoFinalize() bool {
+	// Reuseable bytes ID are always not able to not be finalized
```

Comment:
nit:
`// Reuseable bytes ID can never be finalized as this ID is reused with Reset()`